llvm/lib/CodeGen/GlobalISel/LegalizerHelper.cpp

```diff
@@ -5314,6 +5314,14 @@ LegalizerHelper::moreElementsVector(MachineInstr &MI, unsigned TypeIdx,
     Observer.changedInstr(MI);
     return Legalized;
   }
+  case TargetOpcode::G_ICMP: {
+    Observer.changingInstr(MI);
+    moreElementsVectorSrc(MI, MoreTy, 2);
+    moreElementsVectorSrc(MI, MoreTy, 3);
+    moreElementsVectorDst(MI, MoreTy, 0);
```

Comment:
I'm not sure how much this matters, but this looks like it is assuming that the three types are all the same, not that one is a predicate vector (i1 elements). Neon will work with full vector lane mask results, so it should work OK there, but possible not for other architectures with vector predicate register like SVE and MVE.

Reply:
The `MoreTy` comes from here:
https://github.com/llvm/llvm-project/blob/c1a442462d9d47f86392da7e7019ab218d7a3088/llvm/lib/CodeGen/GlobalISel/LegalizerHelper.cpp#L5155
Properly supporting predicates will take more work.

Reply:
Furthermore, SVE and moreElements is incompatible. I cannot change the length of a scalable vector.

Reply:
Done.